proteuslib/property_models/tests/property_test_harness.py

```diff
@@ -0,0 +1,398 @@
+###############################################################################
+# ProteusLib Copyright (c) 2021, The Regents of the University of California,
+# through Lawrence Berkeley National Laboratory, Oak Ridge National
+# Laboratory, National Renewable Energy Laboratory, and National Energy
+# Technology Laboratory (subject to receipt of any required approvals from
+# the U.S. Dept. of Energy). All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/nawi-hub/proteuslib/"
+#
+###############################################################################
+
+import pytest
+
+from pyomo.environ import (ConcreteModel,
+                           Set,
+                           Param,
+                           Var,
+                           Expression,
+                           Constraint,
+                           value,
+                           units as pyunits,
+                           SolverStatus)
+from pyomo.util.check_units import assert_units_consistent, check_units_equivalent
+from idaes.core import FlowsheetBlock
+from idaes.core.util.model_statistics import (degrees_of_freedom,
+                                              number_variables,
+                                              number_total_constraints,
+                                              number_unused_variables)
+from idaes.core.util.scaling import (calculate_scaling_factors,
+                                     unscaled_variables_generator,
+                                     unscaled_constraints_generator,
+                                     badly_scaled_var_generator)
+from idaes.core.util.exceptions import (PropertyPackageError,
+                                        PropertyNotSupportedError)
+from idaes.core.util import get_solver
+
+# -----------------------------------------------------------------------------
+# Get default solver for testing
+solver = get_solver()
+solver.options["nlp_scaling_method"] = "user-scaling"
```

Comment:
I would suggest that the test harness take arguments for these, so that it can be configured as needed when it is used in a specific test case.

Reply:
Yes. I'll add that as an input.